packages/connectivity/src/scp-cf/destination/destination-accessor-provider-subscriber-lookup.spec.ts

```diff
@@ -347,19 +353,81 @@ describe('call getAllDestinations with and without subscriber token', () => {
     mockGetAllProvider();
 
     const debugSpy = jest.spyOn(logger, 'debug');
+    const retrieveSpy = jest.spyOn(
+      destinationServiceCache,
+      'retrieveDestinationsFromCache'
+    );
+    const cacheSpy = jest.spyOn(
+      destinationServiceCache,
+      'cacheRetrievedDestinations'
+    );
 
-    const allDestinations = await getAllDestinationsFromDestinationService({
+    // First call - should fetch and cache
+    let allDestinations = await getAllDestinationsFromDestinationService({
       jwt: subscriberUserToken
     });
 
     expect(allDestinations).toEqual([parsedSubscriberDestination]);
-
+    expect(retrieveSpy).toHaveBeenCalled();
+    expect(cacheSpy).toHaveBeenCalled();
     expect(debugSpy).toHaveBeenCalledWith(
       'Retrieving all destinations for account: "subscriber" from destination service.'
     );
     expect(debugSpy).toHaveBeenCalledWith(
       'Retrieving subaccount destination: DESTINATION.\n'
     );
+
+    // Reset spies for second call
+    jest.clearAllMocks();
+
+    // Second call - should use cache
+    allDestinations = await getAllDestinationsFromDestinationService({
+      jwt: subscriberUserToken
+    });
+
+    expect(allDestinations).toEqual([parsedSubscriberDestination]);
+    expect(retrieveSpy).toHaveBeenCalled();
+    expect(cacheSpy).not.toHaveBeenCalled(); // Should not cache again
+  });
+
+  it('should fetch all subscriber destinations and not cache destinations', async () => {
+    mockGetAllSubscriber();
+    mockGetAllProvider();
+
+    const cacheSpy = jest.spyOn(
+      destinationServiceCache,
+      'cacheRetrievedDestinations'
+    );
+
+    const retrieveSpy = jest.spyOn(
+      destinationServiceCache,
+      'retrieveDestinationsFromCache'
+    );
+
+    let allDestinations = await getAllDestinationsFromDestinationService({
+      jwt: subscriberUserToken,
+      useCache: false
+    });
+
+    expect(allDestinations).toEqual([parsedSubscriberDestination]);
+    expect(cacheSpy).not.toHaveBeenCalled();
+    expect(retrieveSpy).not.toHaveBeenCalled();
+
+    // Reset spies and mocks for second call
+    jest.clearAllMocks();
+    nock.cleanAll();
+    mockGetAllSubscriber();
+    mockGetAllProvider();
+
+    // Second call - should fetch again since cache was not used previously
```

Comment:
The second setup is not really needed since we already estbalished: expect(cacheSpy).not.toHaveBeenCalled();
expect(retrieveSpy).not.toHaveBeenCalled();

Reply:
Changed the test as per your [suggestion](https://github.com/SAP/cloud-sdk-js/pull/5440#discussion_r1957898898).